TestCase/Managers/GameManager.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace TestCase.Managers
+{
+    public class GameManager
+    {
+        private readonly BoardManager _boardManager;
+        private readonly InputManager _inputManager;
+
+        private bool isPlayerXTurn = true;
+        private bool isPlayerOTurn = false;
+
+        private int turnCounter = 0;
+
+        public GameManager()
+        {
+            this._boardManager = new BoardManager();
+            this._inputManager = new InputManager();
+        }
+
+        public void InputFromUser()
+        {
+            if (isPlayerXTurn)
+            {
+                Console.WriteLine("\nit is currently Player X's turn");
+            }
+            else if (isPlayerOTurn)
+            {
+                Console.WriteLine("\nit is currently Player O's turn");
+            }
+
+            int row = _inputManager.GetInput() - 1;
+
+            int col = _inputManager.GetInput() - 1;
+
+            PlayerTurn(row, col);
+        }
+
+        public void PlayerTurn(int row, int col)
+        {
+           if (_boardManager.PlaceToken(row, col,isPlayerXTurn ? GlobalConsts.X_PLAYER_TOKEN : GlobalConsts.O_PLAYER_TOKEN)) 
+            {
+                isPlayerXTurn = !isPlayerXTurn;
+                isPlayerOTurn = !isPlayerOTurn;
+            }
+            else 
+            {
+                Console.WriteLine("\nSpace taken try again");
+                InputFromUser();
+            }
+        }
+
+        public void ResetGame()
+        {
+            Console.WriteLine("\n++++++++++++++++++++++++++++++NEW GAME++++++++++++++++++++++++++++++++++++++");
+            turnCounter = 0;
+
+            isPlayerXTurn = true;
+            isPlayerOTurn = false;
+
+            _boardManager.ResetBoard();
+
+            StartGame();
+
+        }
+
+        public void StartGame()
+        {
+            while (turnCounter < 9)
+            {
+                _boardManager.PrintBoard();
+
+                InputFromUser();
+
+                turnCounter++;
+
+                if (turnCounter >= 5 && _boardManager.CheckWin())
+                {
+                    _boardManager.PrintBoard();
+                    if (_inputManager.ResetChoiceInput())
```

Comment:
```suggestion
                    _inputManager.ResetChoiceInput() 
                       ? ResetGame()
                       : Environment.Exit(0);
```

Ternary makes this shorter and a bit nicer to look at. Would prefer them over single line if-else statements :D

Reply:
this one was the one that gave me some grief

Reply:
Ah yeah, the methods return void which means it wont work as I expected. Resolved